### PR TITLE
REST endpoint and tests for data stream migration

### PR DESCRIPTION
**Pull request by danhermann**
Provides the REST endpoint for migrating existing indices to data streams.

Relates to #61046

### CODE REVIEW

File: x-pack/plugin/src/test/resources/rest-api-spec/test/data_stream/130_migrate_to_data_stream.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+---
+"Migrate to data stream":
+  - skip:
+      version: " - 7.99.99"
+      reason: "change to 7.10.99 after backport"
```
Comment (martijnvg) [inline x-pack/plugin/src/test/resources/rest-api-spec/test/data_stream/130_migrate_to_data_stream.yml:5]: I think `change to 7.10.99 after backport` should be `change to 7.9.99 after backport`?

**Review by martijnvg — APPROVED** [review]
LGTM

File: x-pack/plugin/src/test/resources/rest-api-spec/test/data_stream/130_migrate_to_data_stream.yml (hunk 1 above)
Comment (danhermann) [reply x-pack/plugin/src/test/resources/rest-api-spec/test/data_stream/130_migrate_to_data_stream.yml:5]: I don't think the migrate feature will be available until the 7.11 release?

Comment (martijnvg) [reply x-pack/plugin/src/test/resources/rest-api-spec/test/data_stream/130_migrate_to_data_stream.yml:5]: 🤦 true, I somehow always think the feature makes it into the 7.10 release :)

**Pull request merged**